Remove InjectedBundlePagePolicyClient

achristensen07 opened the pull request:
#### 9eeab8da91f070e26e251a4e8b853e844eeb2410
<pre>
Remove InjectedBundlePagePolicyClient
<a href="https://bugs.webkit.org/show_bug.cgi?id=247169">https://bugs.webkit.org/show_bug.cgi?id=247169</a>
rdar://101655797

Reviewed by Tim Horton.

There was one layout test, fast/loader/policy-delegate-action-hit-test-zoomed.html,
which printed the dom node path of the clicked link.  This functionality is not available
nor needed in WK2.  The WK1 test still verifies it.  WK2 has _WKHitTestResult with tests
to verify we get the node info we need in this case.

* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetPolicyClient):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePagePolicyClient.cpp: Removed.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePagePolicyClient.h: Removed.
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForResponse):
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNewWindowAction):
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction):
(WebKit::WebFrameLoaderClient::dispatchUnableToImplementPolicy):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::close):
(WebKit::WebPage::initializeInjectedBundlePolicyClient): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::injectedBundleLoaderClient):
(WebKit::WebPage::injectedBundlePolicyClient): Deleted.

Canonical link: <a href="https://commits.webkit.org/256491@main">https://commits.webkit.org/256491@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bcb224221ff30ee95c0530a5ffed195bf7664c62

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95757 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5011 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28799 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105335 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165642 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5091 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33768 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88138 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101168 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101418 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3745 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82370 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30798 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/85607 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87516 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73627 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39502 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/19054 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37195 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20371 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41259 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/43012 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2162 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43273 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39623 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on LayoutTests/fast/loader/onload-policy-ignore-for-frame-async-delegates.html:
```diff
@@ -6,6 +6,7 @@
         if (window.testRunner) {
             testRunner.dumpAsText();
             testRunner.setCustomPolicyDelegate(true);
+            if (testRunner.skipPolicyDelegateNotifyDone) { testRunner.skipPolicyDelegateNotifyDone() }
```
Why the weird style? Should be on the next line with no curly braces, just like in C++

achristensen07 replied:
JS is slightly different than C++ in its handling of if statements with/without {} and I like to do this to be as unambiguous as possible.

Commit:
Remove InjectedBundlePagePolicyClient
https://bugs.webkit.org/show_bug.cgi?id=247169
rdar://101655797

Reviewed by Tim Horton.

There was one layout test, fast/loader/policy-delegate-action-hit-test-zoomed.html,
which printed the dom node path of the clicked link.  This functionality is not available
nor needed in WK2.  The WK1 test still verifies it.  WK2 has _WKHitTestResult with tests
to verify we get the node info we need in this case.

* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetPolicyClient):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePagePolicyClient.cpp: Removed.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePagePolicyClient.h: Removed.
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForResponse):
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNewWindowAction):
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction):
(WebKit::WebFrameLoaderClient::dispatchUnableToImplementPolicy):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::close):
(WebKit::WebPage::initializeInjectedBundlePolicyClient): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::injectedBundleLoaderClient):
(WebKit::WebPage::injectedBundlePolicyClient): Deleted.

Canonical link: https://commits.webkit.org/256491@main
